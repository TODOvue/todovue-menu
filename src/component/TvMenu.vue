<template>
  <header class="tv-menu-container">
    <nav class="tv-menu-image tv-cursor-pointer">
      <img :src="imageMenu" alt="Logo" @click="handleClickImage" />
    </nav>
    <section class="tv-menu-items">
      <ul v-for="menu in menus" :key="menu.id">
        <li
          class="tv-menu-item tv-cursor-pointer"
          @click="handleClickMenu(menu)"
        >
          {{ menu.title }}
        </li>
      </ul>
      <i
        class="tv-icon tv-icon-position-right tv-menu-icon tv-cursor-pointer"
        @click="manageViewMenu(true)"
      ></i>
      <tv-search
        :placeholder="placeholder"
        :titleButton="titleButton"
        @search="handleSearch"
      />
      <div class="tv-menu-items-mobile" v-if="showMenu">
        <i
          class="tv-icon tv-menu-icon-cancel tv-cursor-pointer"
          @click="manageViewMenu(false)"
        ></i>
        <ul v-for="menu in menus" :key="menu.id">
          <li
            class="tv-menu-item-mobile tv-cursor-pointer"
            @click="handleClickMenu(menu)"
          >
            {{ menu.title }}
          </li>
        </ul>
      </div>
    </section>
  </header>
</template>

<script>
import TvSearch from "todovue-search";
import { ref } from "vue";
export default {
  name: "TvMenu",
  components: {
    TvSearch,
  },
  props: {
    placeholder: {
      type: String,
      default: "",
    },
    titleButton: {
      type: String,
      default: "",
    },
    menus: {
      type: Array,
      default: () => [],
    },
    imageMenu: {
      type: String,
      default: "",
    },
  },
  setup(_, { emit }) {
    const showMenu = ref(false);

    const handleClickImage = () => {
      emit("clickImage");
    };

    const handleClickMenu = (menu) => {
      if (showMenu.value) {
        manageViewMenu(false);
      }
      emit("clickMenu", menu);
    };

    const handleSearch = (search) => {
      emit("searchMenu", search);
    };

    const manageViewMenu = (value) => {
      showMenu.value = value;
    };

    return {
      showMenu,
      handleClickImage,
      handleClickMenu,
      handleSearch,
      manageViewMenu,
    };
  },
  emits: ["clickImage", "clickMenu", "searchMenu"],
};
</script>

<style></style>
