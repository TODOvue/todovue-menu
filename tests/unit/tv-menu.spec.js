import { shallowMount } from "@vue/test-utils";
import TvMenu from "@/component/TvMenu.vue";

describe("TvMenu", () => {
  const menus = [
    { id: 1, title: "Home" },
    { id: 2, title: "About" },
    { id: 3, title: "Blog" },
  ];

  it("should matches snapshot", () => {
    const wrapper = shallowMount(TvMenu, {
      propsData: {
        menus,
        imageMenu: "image.jpg",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('emits the "clickImage" event when the image is clicked', () => {
    const wrapper = shallowMount(TvMenu, {
      propsData: {
        menus,
        imageMenu: "image.jpg",
      },
    });
    wrapper.find(".tv-menu-image img").trigger("click");
    expect(wrapper.emitted("clickImage")).toBeTruthy();
  });

  it('emits the "clickMenu" event with the corresponding option when one of the menu options is clicked', () => {
    const wrapper = shallowMount(TvMenu, {
      propsData: {
        menus,
        imageMenu: "image.jpg",
      },
    });
    wrapper.find(".tv-menu-item:first-of-type").trigger("click");
    expect(wrapper.emitted("clickMenu")).toBeTruthy();
    expect(wrapper.emitted("clickMenu")[0][0]).toEqual(menus[0]);
  });

  it('emits the "searchMenu" event when a search is performed', () => {
    const wrapper = shallowMount(TvMenu);
    const searchBox = wrapper.findComponent({ name: "TvSearch" });

    searchBox.vm.$emit("search", "blogs");

    expect(wrapper.emitted("searchMenu")).toBeTruthy();
    expect(wrapper.emitted("searchMenu")[0]).toEqual(["blogs"]);
  });
});
