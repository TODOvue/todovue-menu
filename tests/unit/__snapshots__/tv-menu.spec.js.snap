// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvMenu should matches snapshot 1`] = `
<header class="tv-menu-container">
  <nav class="tv-menu-image tv-cursor-pointer"><img src="image.jpg" alt="Logo"></nav>
  <section class="tv-menu-items">
    <ul>
      <li class="tv-menu-item tv-cursor-pointer">Home</li>
    </ul>
    <ul>
      <li class="tv-menu-item tv-cursor-pointer">About</li>
    </ul>
    <ul>
      <li class="tv-menu-item tv-cursor-pointer">Blog</li>
    </ul><i class="tv-icon tv-menu-icon tv-cursor-pointer"></i>
    <tv-search-stub placeholder="" titlebutton="" results=""></tv-search-stub>
    <!--v-if-->
  </section>
</header>
`;
